src/config.rs

```diff
@@ -129,59 +129,98 @@ impl Config {
                     slot.dbtype = Some(typ_.to_string());
                     slot.dbpath = Some(name.to_string());
                     /* if this fails there will be no slots defined */
-                    let _ = conf.add_slot(slot);
+                    let _ = conf.slots.push(slot);
                 }
                 Err(_) => (),
             }
         }
-        return Ok(conf);
+        Ok(conf)
+    }
+
+    fn fix_slot_numbers(&mut self) {
+        let mut slotnum: u32 = 0;
+        /* if there are any slot missing a valid slot number
+         * we are going to allocate slots numbers after the highest
+         * properly configured one. Note that the config file format
+         * requires slot numbers, so this generally happens for legacy
+         * or init args configurations only, ie a single slot */
+        let mut missing = false;
+        for slot in &self.slots {
+            if slot.slot != u32::MAX {
+                if slotnum <= slot.slot {
+                    slotnum += slot.slot + 1;
```

Comment:
I think this should be just
```
                    slotnum = slot.slot + 1;
```
otherwise you do not use the first unused, but sum of all unused + some.